bt_ddos_shield/state_manager.py

```diff
@@ -1,78 +1,473 @@
 from abc import ABC, abstractmethod
+from collections import defaultdict
 from datetime import datetime
-from enum import Enum
+from types import MappingProxyType
+from typing import Optional, Union
+from sqlalchemy import create_engine, Column, String, DateTime, ForeignKey, Integer, CheckConstraint, Engine, \
+    PrimaryKeyConstraint
+from sqlalchemy.engine import url
+from sqlalchemy.orm import sessionmaker, DeclarativeBase
 
-from bt_ddos_shield.address import Address
-from bt_ddos_shield.miner_shield import Hotkey
+from bt_ddos_shield.address import Address, AddressType
+from bt_ddos_shield.utils import Hotkey, PublicKey
 
 
-class MinerShieldPhase(Enum):
-    """
-    Possible phases of shield.
-    """
-    DISABLED = "disabled"                         # disabled - initial state
-    MANIFEST_PUBLISHED = "manifest_published"     # manifest is saved to storage
-    MANIFEST_BROADCASTED = "manifest_broadcasted" # info about manifest is published to blockchain
-    ENABLED = "enabled"                           # shield is enabled
+class MinerShieldState:
+    _known_validators: dict[Hotkey, PublicKey]
+    _banned_validators: dict[Hotkey, datetime]
+    _validators_addresses: dict[Hotkey, Address]
+    _manifest_address: Optional[Address]
+    _address_manager_state: dict[str, str]
+    _address_manager_created_objects: dict[str, frozenset[str]]
 
+    def __init__(self, known_validators: dict[Hotkey, PublicKey], banned_validators: dict[Hotkey, datetime],
+                 validators_addresses: dict[Hotkey, Address], manifest_address: Optional[Address],
+                 address_manager_state: dict[str, str],
+                 address_manager_created_objects: dict[str, frozenset[str]]):
+        super().__setattr__('_known_validators', known_validators)
```

Comment:
why super when there's no superclass?

Reply:
I made this class immutable by overloading __setattr__ method. That's why I need to call setattr from super class to set these attributes. I asked copilot about this and got response, that this initialization can be done other way: `self.__dict__['_known_validators'] = known_validators`. I'm not sure if it is better than calling setattr from super class. Tell me :-)